specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2020-06-01-preview/rbac.json

```diff
@@ -0,0 +1,552 @@
+        "x-ms-examples": {
+          "CosmosDBSqlRoleDefinitionCreateUpdate": {
+            "$ref": "./examples/CosmosDBSqlRoleDefinitionCreateUpdate.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "description": "Creates or updates an Azure Cosmos DB Role Definition.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/roleDefinitionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/databaseAccountNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "requestBody",
+            "in": "body",
+            "required": true,
+            "description": "The properties required to create a new role definition.",
+            "schema": {
+              "$ref": "#/definitions/SqlRoleDefinitionProperties"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The Role Definition create or update request was accepted and will complete asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/SqlRoleDefinitionResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "operationId": "SqlRoleDefinition_Delete",
+        "x-ms-examples": {
+          "CosmosDBSqlRoleDefinitionDelete": {
+            "$ref": "./examples/CosmosDBSqlRoleDefinitionDelete.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "description": "Deletes an existing Azure Cosmos DB Role Definition.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/roleDefinitionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/databaseAccountNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "The Role Definition was deleted.",
+            "schema": {
+              "$ref": "#/definitions/SqlRoleDefinitionResponse"
+            }
+          },
+          "204": {
+            "description": "Role Definition does not exist"
+          }
```

Comment:
404 is for Not found. In case of Delete 204 is returned because it indicates that the operation succeeded but there is no content to be shown. Please change the description. Error code is correct. (same for the above 202, error code is correct, description is not)

Reply:
thanks for the explanation! fixed

Reply:
Just to be clear, 204 is for the case when resource on which delete request is being sent, doesnt exist.